test/manyvariables.jl

```diff
@@ -333,6 +337,36 @@ using Base.Test
     hessian!(hes2,g1(xT+1,yT-1)-g2(xT+1,yT-1))
     @test hes1 == hes2
 
+    a = 3x + 4y +6x^2 + 8x*y
+    @test typeof( norm(x) ) == Float64
+    @test norm(x) > 0
+    @test norm(a) == norm([3,4,6,8.0])
+    @test norm(a,4) == sum([3,4,6,8.0].^4)^(1/4.)
+    @test norm(a,Inf) == 8.
+    @test norm((3.+4im)*x) == abs(3.+4im)
```

Comment:
I am checking what is going on with travis for julia 0.7-dev on mac, and on doing this, I just noticed that julia 0.7 complains about `3.+4im` as ambiguous (due to parsing conflicts, essentially to know if you mean `3. + ...` or `3 .+ ...`). I suggest to write it as `3.0 + 4im`. Can you change it? See https://github.com/JuliaLang/julia/issues/19089.